js/StickySidebarComponent.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+var $ = require('jquery');
+
+class StickySidebarComponent {
+
+    /**
+     * StickySidebarComponent
+     * @constructor 
+     * @param {object} window - window object
+     * @param {jQuery} $html - jQuery wrapper of the html node
+     */
+    constructor ($html, window) {
+        this.$html = $html;
+        this.window = window;
+    }
+
+    /**
+     * Initialise component
+     * 
+     * Should be fired on page load, checks for required options, performs an 
+     * initial sticky calculation and binds to resize & scroll events
+     */
+    init () {
+        if (typeof this.$html === 'undefined' || !this.$html) {
+            throw new Error('$html must be passed to StickySidebarComponent');
+        }
+
+        if (typeof this.window === 'undefined' || !this.window) {
+            throw new Error('window must be passed to StickySidebarComponent');
+        }
+
+        this.sticky()
+        $(this.window).on('scroll resize', (event) => this.sticky(event));
+    }
+
+    /**
+     * Sticky calculation
+     * 
+     * If content is taller than settings, and the viewport has been scrolled 
+     * enough then add the required sticky class to the container, and remove it
+     * if no longer required
+     */
+    sticky () {
+        const $container = this.$html.find('.tab__container.has-settings .tab__inner');
+
+        if (!$container.length) {
+            return false;
```

Comment:
why return false here?

Reply:
Stop any errors being thrown if sticky() is used but the correct layout is not present

Reply:
I understand the early return, but why return the value `false` ? should this not just be 

```
if (!...) {
    return;
}
```